### PR TITLE
Rename to AzureSpringApps

### PR DESCRIPTION
**Pull request by yitaopan**
**Task name**: AzureSpringCloudV0

**Description**: Rename Azure Spring Cloud too Azure Spring Apps (the new name of this service), keep task name so this task can still be referenced.

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AzureSpringCloudV0/deploymentProvider/AzureSpringCloudDeploymentProvider.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class AzureSpringCloudDeploymentProvider {
         var azureSpringCloudResourceId: string;
         if (this.taskParameters.AzureSpringCloud.startsWith('/')) {
             if (this.taskParameters.AzureSpringCloud.includes('..')){{
-                throw Error(tl.loc('InvalidAzureSpringCloudResourceId', 'this.taskParameters.AzureSpringCloud'));
+                throw Error(tl.loc('InvalidAzureSpringAppsResourceId', 'this.taskParameters.AzureSpringCloud'));
```
Comment (bowen5) [inline Tasks/AzureSpringCloudV0/deploymentProvider/AzureSpringCloudDeploymentProvider.ts:33]: Do we need to change "this.taskParameters.AzureSpringCloud"?

File: Tasks/AzureSpringCloudV0/operations/taskparameters.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Package, PackageType } from 'azure-pipelines-tasks-webdeployment-common
 
 export class Inputs {
     public static readonly connectedServiceName = 'ConnectedServiceName';
-    public static readonly azureSpringCloud = 'AzureSpringCloud';
+    public static readonly azureSpringCloud = 'AzureSpringApps';
```
Comment (bowen5) [inline Tasks/AzureSpringCloudV0/operations/taskparameters.ts:5]: If won't introduce breaking, how about change this param name "azureSpringCloud" as well?

**Review by bowen5 — CHANGES_REQUESTED** [review]

File: Tasks/AzureSpringCloudV0/operations/taskparameters.ts (hunk 2 above)
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/operations/taskparameters.ts:5]: Yes, this could introduce breaking, I will change it back.

File: Tasks/AzureSpringCloudV0/package-lock.json [hunk 3]
```diff
@@ -468,7 +468,7 @@
         "xmldom": "^0.1.27"
       }
     },
-    "node_modules/azure-pipelines-tasks-webdeployment-common-v4/node_modules/@types/mocha": {
+    "node_modules/azure-pipelines-tasks-webdeployment-common/node_modules/@types/mocha": {
```
Comment (bowen5) [inline Tasks/AzureSpringCloudV0/package-lock.json:471]: Just double confirm: it is by design to drop the "v4" instead of make it "v5", right?

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts [hunk 4]
```diff
@@ -246,6 +246,10 @@ export class AzureSpringCloud {
                     throw ToError(response);
                 }
                 provisioningState = response.body.properties.provisioningState;
+                if (provisioningState == "Failed"){
```
Comment (bowen5) [inline Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts:249]: Good catch.

File: Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@
 import tl = require('azure-pipelines-task-lib/task');
 import { TaskParameters, TaskParametersUtility } from './operations/taskparameters';
 
-import { AzureSpringCloudDeploymentProvider } from './deploymentProvider/AzureSpringCloudDeploymentProvider'
+import { AzureSpringAppsDeploymentProvider } from './deploymentProvider/AzureSpringCloudDeploymentProvider'
```
Comment (bowen5) [inline Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts:5]: Are we able to change "AzureSpringCloudDeploymentProvider"? Probably need to renam the file .

**Review by bowen5 — CHANGES_REQUESTED** [review]

File: Tasks/AzureSpringCloudV0/package-lock.json (hunk 3 above)
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/package-lock.json]: There might be some conflict, let me double check.

File: Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts (hunk 5 above)
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts:5]: Ok, will do, I was about to check with you whether this is needed.

Comment (bowen5) [reply Tasks/AzureSpringCloudV0/azurespringclouddeployment.ts:5]: Thanks, good to know that.

**Comment by bowen5** [comment]
LGTM.

**Review by bowen5 — APPROVED** [review]

**Review by bowen5 — APPROVED** [review]

File: Tasks/AzureSpringCloudV0/operations/taskparameters.ts [hunk 6]
```diff
@@ -50,7 +51,7 @@ export class TaskParametersUtility {
         console.log("global['_vsts_task_lib_loaded'] = " + global['_vsts_task_lib_loaded']);
         var taskParameters: TaskParameters = {
             ConnectedServiceName: tl.getInput(Inputs.connectedServiceName, true),
-            AzureSpringCloud: tl.getInput(Inputs.azureSpringCloud, true),
```
[the remark this answers is not in the document]
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/operations/taskparameters.ts:53]: Hi, @kirill-ivlev thanks for pointing out! To my understanding, the input of this parameter is on line 7 of this file, where I kept the original name, please correct me if I am wrong.

[the remark this answers is not in the document]
Comment (kirill-ivlev) [reply Tasks/AzureSpringCloudV0/operations/taskparameters.ts:53]: You right, thanks, LGTM

**Review by kirill-ivlev — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Comment by bowen5** [comment]
@kirill-ivlev @vmapetr @mmrazik can any of you help to merge? Thanks.

**Comment by KonstantinTyukalov** [comment]
@bowen5 please, resolve PR conflicts.
Also, it seems you have some issues with dependencies files:
https://dev.azure.com/mseng/PipelineTools/_build/results?buildId=19626785&view=logs&j=55b94f1c-9d32-5114-bc90-83768ebbe0df&t=8334df59-c1ae-533d-4d46-13427016ffd1&l=18

```text
> git diff --name-only /home/vsts/work/1/s/Tasks/AzureSpringCloudV0
Tasks/AzureSpringCloudV0/Tests/package-lock.json
Tasks/AzureSpringCloudV0/package-lock.json

Uncommitted changes found:

 - Tasks/AzureSpringCloudV0/Tests/package-lock.json

 - Tasks/AzureSpringCloudV0/package-lock.json


Please build your tasks locally and commit specified changes:

 - Tasks/AzureSpringCloudV0/Tests/package-lock.json
 - Tasks/AzureSpringCloudV0/package-lock.json

Make sure you are using Node 10 and NPM 6. For more details check our contribution guide - ***/blob/master/docs/contribute.md
```

**Comment by yitaopan** [comment]
@KonstantinTyukalov Hi, thanks, I have merged master branch and rebuild the task using Node 10 and NPM 6. Can you please take a look again?

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts [hunk 7]
```diff
@@ -246,6 +246,10 @@ export class AzureSpringCloud {
                     throw ToError(response);
                 }
                 provisioningState = response.body.properties.provisioningState;
+                if (provisioningState == "Failed"){
+                    tl.debug('KPack build failed');
+                    throw ToError(response);
+                }
```
Comment (KonstantinTyukalov) [inline Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts:252]: [Suggestion] These changes aren't seems related to rebranding 😃
Probably, make sense to move it into separate PR, but it's up to you

**Review by KonstantinTyukalov — APPROVED** [review]
LGTM, but please check few comments and bump task version according to the current sprint

File: Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts (hunk 7 above)
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-apps.ts:252]: Thanks, sorry I didn't pointed this out, this would be a small ux improvement I think I could include here.

File: Tasks/AzureSpringCloudV0/package.json [hunk 8]
```diff
@@ -25,14 +25,15 @@
     "@types/mocha": "8.2.1",
     "@types/node": "^10.17.0",
     "@types/q": "1.0.7",
-    "JSONPath": "^0.11.2",
+    "@xmldom/xmldom": "^0.8.4",
```
[the remark this answers is not in the document]
Comment (yitaopan) [reply Tasks/AzureSpringCloudV0/package.json]: I updated the version to 0.8.7 now

**Comment by yitaopan** [comment]
@KonstantinTyukalov I bumped the task version and updated the @xmldom/xmldom to the latest of 0.8.x, can you please take a look again? Thanks
